docs: Update Transform Data (#71013)
grafana/grafana: docs/sources/panels-visualizations/query-transform-data/transform-data/index.md

1. Click the **Transform** tab.
1. Click the trash icon next to the transformation you want to delete.

> 

Reviewer: ```suggestion
```

- **Count** - For non-time related series, use this option to define the number of elements in a bucket.

#### Y Bucket

This setting determines how the y-axis is split into buckets.

#### Y Bucket scale

Use this option to set the scale of the y-axes. Select from:

- **Linear**
- **Logarithmic** - Use a base 2 or base 10.
- **Symlog** - A symmetrical logarithmic scale. Use a base 2 or base 10; allows negative values.

### Extract fields

Use this transformation to select one source of data and extract content from it in different formats. Set the following fields:

- **Source** - Select the field for the source of data.
- **Format** - Select one of the following:
  - **JSON** - To parse JSON content from the source.
  - **Key+value parse** - To parse content in the format `a=b` or `c:d` from the source.
  - **Auto** - To discover fields automatically.
- **Replace all fields** - Optional: Select this option if you want to hide all other fields and display only your calculated field in the visualization.
- **Keep time** - Optional: Only available if **Replace all fields** is true. Keep the time field in the output.

Consider the following data set:

> | timestamp           | json_data     |

Reviewer: capitalizing the column headers where appropriate to be consistent with existing tables on this page
```suggestion
| Timestamp           | json_data     |
```

| ------------------- | ------------- |
| 1636678740000000000 | {"value": 1}  |
| 1636678680000000000 | {"value": 5}  |
| 1636678620000000000 | {"value": 12} |

> You could prepare the data to be used by a [Time series panel]({{< relref "../../visualizations/time-series" >}}) with this config:

Reviewer: ```suggestion
You could prepare the data to be used by a [Time series panel]({{< relref "../../visualizations/time-series" >}}) with this configuration:
```

- Source: json_data
- Format: JSON
  - Field: value
  - alias: my_value
- Replace all fields: true
- Keep time: true

> This will output:

Reviewer: ```suggestion
This will generate the following output:
```

> | timestamp           | my_value |

Reviewer: ```suggestion
| Timestamp           | my_value |
```

| ------------------- | -------- |
| 1636678740000000000 | 1        |
| 1636678680000000000 | 5        |
| 1636678620000000000 | 12       |

### Field lookup

Use this transformation on a field value to look up additional fields from an external source.

- **Field** - Select a text field.
- **Lookup** - Select from **Countries**, **USA States**, and **Airports**.

This transformation currently supports spatial data.

> For example, if we have this data:

Reviewer: changing to preferred second person 
```suggestion
For example, if you have this data:
```

> | location  | values |

Reviewer: ```suggestion
| Location  | Values |
```

| --------- | ------ |
| AL        | 0      |
| AK        | 10     |
| Arizona   | 5      |
| Arkansas  | 1      |
| Somewhere | 5      |

> With this config:

Reviewer: ```suggestion
With this configuration:
```

- Field: location
- Lookup: USA States

> We will get this output:

Reviewer: ```suggestion
You'll get the following output:
```

> | location  | id  | name     | lng         | lat       | values |

Reviewer: ```suggestion
| Location  | ID  | Name     | Lng         | Lat       | Values |
```

| 2019-01-01 00:00:00 | below | 29    |
| 2020-01-01 00:00:00 | above | 22    |

### Create heatmap

> Use this transformation to prepare histogram data to be visualized over time. Similar to the [Heatmap panel] ({{< relref "../visualizations/heatmap" >}}), this transformation allows you to convert histogram metrics to buckets over time.

Reviewer: fixing link
```suggestion
Use this transformation to prepare histogram data to be visualized over time. Similar to the [Heatmap panel]({{< relref "../../visualizations/heatmap" >}}), this transformation allows you to convert histogram metrics to buckets over time.
```

#### X Bucket

This setting determines how the x-axis is split into buckets. 

- **Size** - Specify a time interval in the input field. For example, a time range of `1h` makes the cells one hour wide on the x-axis. 
- **Count** - For non-time related series, use this option to define the number of elements in a bucket.

#### Y Bucket

This setting determines how the y-axis is split into buckets.

#### Y Bucket scale

Use this option to set the scale of the y-axes. Select from:

- **Linear**

> - **Logarithmic** - Use a base 2 or base 10
> - **Symlog** - A symmetrical logarithmic scale - Use a base 2 or base 10; allows negative values

Reviewer: ```suggestion
- **Logarithmic** - Use a base 2 or base 10.
- **Symlog** - A symmetrical logarithmic scale. Use a base 2 or base 10; allows negative values.
```

### Extract fields

Use this transformation to select one source of data and extract content from it in different formats. Set the following fields:

- **Source** - Select the field for the source of data.
- **Format** - Select one of the following:

>   - **JSON** To parse JSON content from the source
>   - **Key+value parse** To parse content in the format a=b or c:d from the source
>   - **Auto** To discover fields automatically

Reviewer: ```suggestion
  - **JSON** -  To parse JSON content from the source.
  - **Key+value parse** -  To parse content in the format `a=b` or `c:d` from the source.
  - **Auto** - To discover fields automatically.
```

> - **Replace all fields** -  (Optional) Select this option if you want to hide all other fields and display only your calculated field in the visualization.

Reviewer: ```suggestion
- **Replace all fields** -  Optional: Select this option if you want to hide all other fields and display only your calculated field in the visualization.
```

### Field lookup

Use this transformation on a field value to look up additional fields from an external source.

- **Field** - Select a text field.

> - **Lookup** - Select from **Countries**, **USA States**, and **Airports**)

Reviewer: ```suggestion
- **Lookup** - Select from **Countries**, **USA States**, and **Airports**.
```

### Grouping to matrix

Use this transformation to combine three fields that will be used as input for Column, Row and Cell value, from the query output and generate a Matrix. This matrix will be calculated as follows:

Given the original data

| Server ID | CPU Temperature | Server Status |
| --------- | --------------- | ------------- |
| server 1  | 82              | Ok            |
| server 2  | 88.6            | OK            |
| server 3  | 59.6            | Shutdown      |

We can generate a Matrix using the values of `Server Status` as colum names, the `Server ID` values as rows name and the `CPU Temperatue` as content of each cell. The content of each cell will apear for the existing column (Server Statues) and row combination (Server ID), for the rest of the cells we can select which value to display between: Null, True, False or Empty.

Thus, the output is:

| Server ID\Server Status | OK    | Shutdown |
| ----------------------- | ----- | -------- |
| server 1                | 82    |          |
| server 2                | 88.6  |          |
| server 3                |       | 59.6     |

### Histogram

> Use this transformation to generate a histogram based on the input data. You can select the bucket size as the distance between the lowest item in the bucket (xMin) and the highest item in the bucket (xMax), and its offset (for non-zero based bucket).

Reviewer: we should have a description of 
```suggestion
Use this transformation to generate a histogram based on the input data. 

- **Bucket size** - The distance between the lowest item in the bucket (xMin) and the highest item in the bucket (xMax).
- **Bucket offset** - The offset for non-zero based buckets.
- **Combine series** - Create a histogram using all the available series.

```

> 
> Additionally, you can toggle the Combine Series option to create an Histogram using all the available series.
> 

Reviewer: ```suggestion
```

This transformation enables you to extract key information from your time series and display it in a convenient way.

### Grouping to matrix

> Use this transformation to combine three fields, that will be used as input for the **Column**, **Row**, and **Cell value** fields, from the query output and generate a matrix. This matrix will be calculated as follows:

Reviewer: ```suggestion
Use this transformation to combine three fields—that will be used as input for the **Column**, **Row**, and **Cell value** fields—from the query output, and generate a matrix. This matrix will be calculated as follows:
```

| server 1  | 82              | OK            |
| server 2  | 88.6            | OK            |
| server 3  | 59.6            | Shutdown      |

We can generate a matrix using the values of `Server Status` as column names, the `Server ID` values as row names, and the `CPU Temperature` as content of each cell. The content of each cell will appear for the existing column (`Server Status`) and row combination (`Server ID`). For the rest of the cells, you can select which value to display between: **Null**, **True**, **False**, or **Empty**.

**Output**

| Server ID\Server Status | OK   | Shutdown |
| ----------------------- | ---- | -------- |
| server 1                | 82   |          |
| server 2                | 88.6 |          |
| server 3                |      | 59.6     |

### Histogram

Use this transformation to generate a histogram based on the input data.

- **Bucket size** - The distance between the lowest item in the bucket (xMin) and the highest item in the bucket (xMax).
- **Bucket offset** - The offset for non-zero based buckets.
- **Combine series** - Create a histogram using all the available series.

**Original data**

> Serie 1:

Reviewer: ```suggestion
Series 1:
```

| A   | B   | C   |
| --- | --- | --- |
| 1   | 3   | 5   |
| 2   | 4   | 6   |
| 3   | 5   | 7   |
| 4   | 6   | 8   |
| 5   | 7   | 9   |

> Serie 2:

Reviewer: ```suggestion
Series 2:
```

{{< figure src="/static/img/docs/transformations/join-fields-after-7-0.png" class="docs-image--no-shadow" max-width= "1100px" >}}

> ### Join by Labels

Reviewer: ```suggestion
### Join by labels
```

Use this transformation to join multiple results into a single table. This is especially useful for converting multiple
time series results into a single wide table with a shared Label field.

> Select the label to `Join` by between the labels available or common accross all time series, and the `Value` name for the output result. 

Reviewer: When you say "the labels available or common across all time series" - are these the same thing or two different sets of labels?
```suggestion

- **Join** - Select the label to join by between the labels available or common across all time series.
- **Value** - The name for the output result. 
```

{{< figure src="/static/img/docs/transformations/join-fields-after-7-0.png" class="docs-image--no-shadow" max-width= "1100px" >}}

### Join by labels

Use this transformation to join multiple results into a single table. This is especially useful for converting multiple

> time series results into a single wide table with a shared Label field.

Reviewer: ```suggestion
time series results into a single wide table with a shared **Label** field.
```
